### PR TITLE
Fix centroid sampling error, centroid should be adjusted when center …

### PR DESCRIPTION
**Pull request by jiaolu**
…sampling is also enabled

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3f0745f37689f175300bb2e2289d8ac58d071fa3
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev eaecf6b9ad7bc3d310e752528f84fd52fba23747</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3f0745f37689f175300bb2e2289d8ac58d071fa3
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev eaecf6b9ad7bc3d310e752528f84fd52fba23747</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3f0745f37689f175300bb2e2289d8ac58d071fa3
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JacobHeAMD** [comment]
Hi @jiaolu , please rebase the pull request to the latest dev code, there is a change to add an option to enable lit test. Jenkins test can't pass without it.

**Comment by jiaolu** [comment]
> Hi @jiaolu , please rebase the pull request to the latest dev code, there is a change to add an option to enable lit test. Jenkins test can't pass without it.

yes, rebase and push again

**Comment by amdvlk-admin** [comment]
### Test summary for commit 1748730abd533244a90d61218faef41d846477b6
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

File: patch/llpcPatchInOutImportExport.cpp [hunk 1]
```diff
@@ -6701,13 +6701,14 @@ Value* PatchInOutImportExport::AdjustCentroidIJ(
 {
     auto& entryArgIdxs = m_pContext->GetShaderInterfaceData(ShaderStageFragment)->entryArgIdxs.fs;
     auto pPrimMask = GetFunctionArgument(m_pEntryPoint, entryArgIdxs.primMask);
-    auto pPipelineInfo = static_cast<const GraphicsPipelineBuildInfo*>(m_pContext->GetPipelineBuildInfo());
+    auto& builtInUsage = m_pContext->GetShaderResourceUsage(ShaderStageFragment)->builtInUsage.fs;
     Value* pIJ = nullptr;
 
-    if(pPipelineInfo->rsState.numSamples <= 1)
+    if(builtInUsage.centroid && builtInUsage.center)
```
Comment (amdrexu) [inline patch/llpcPatchInOutImportExport.cpp]: if( -> if (

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 021dc00ebe94fd241881c6e80c246c31f31cf3ac
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 259454/259455 (100.0%)</li>
<li><details><summary>Failed: 1/259455 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.renderpass.suballocation.multisample_resolve.layers_3.b8g8r8a8_srgb.samples_8
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 375327/375327 (100.0%)</li>
<li>Failed: 0/375327 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 021dc00ebe94fd241881c6e80c246c31f31cf3ac
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 294867/294868 (100.0%)</li>
<li><details><summary>Failed: 1/294868 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.image.swapchain_mutable.wayland.2d.r8g8b8a8_unorm_r8g8b8a8_snorm_clear_copy_format_list
Stack trace: Failed to open display at tcuLnxWayland.cpp:82
```
</p></details></li>


</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 021dc00ebe94fd241881c6e80c246c31f31cf3ac
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 021dc00ebe94fd241881c6e80c246c31f31cf3ac
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 69f94c826396761c3cb9003b10af87e76aabf49f
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 294147/294148 (100.0%)</li>
<li><details><summary>Failed: 1/294148 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.image.swapchain_mutable.xlib.2d.r8g8b8a8_unorm_r8g8b8a8_snorm_clear_copy_format_list
Stack trace: Failed to open display: '' at tcuLnxX11.cpp:67
```
</p></details></li>


</ul></details>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 69f94c826396761c3cb9003b10af87e76aabf49f
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 08fcbd5ab2f1cc775f887ffeb8f4cd3ee7ccc31b
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**